dm/worker/server.go

```diff
@@ -749,3 +780,85 @@ func makeCommonWorkerResponse(reqErr error) *pb.CommonWorkerResponse {
 	}
 	return resp
 }
+
+// all subTask in subTaskCfgs should have same source
+// this function return the min position in all subtasks, used for relay's position
+func getMinPosInAllSubTasks(subTaskCfgs []*config.SubTaskConfig) (minPos *mysql.Position, err error) {
+	for _, subTaskCfg := range subTaskCfgs {
+		pos, err := getMinPosForSubTaskFunc(subTaskCfg)
+		if err != nil {
+			return nil, err
+		}
+
+		if pos == nil {
+			continue
+		}
+
+		if minPos == nil {
+			minPos = pos
+		} else {
+			if minPos.Compare(*pos) >= 1 {
+				minPos = pos
+			}
+		}
+	}
+
+	return minPos, nil
+}
+
+func getMinPosForSubTask(subTaskCfg *config.SubTaskConfig) (minPos *mysql.Position, err error) {
+	dbCfg := subTaskCfg.To
+	dbCfg.RawDBCfg = config.DefaultRawDBConfig().SetReadTimeout("10s")
+	db, err := conn.DefaultDBProvider.Apply(dbCfg)
+	if err != nil {
+		return nil, terror.WithScope(err, terror.ScopeDownstream)
+	}
+	defer db.Close()
+
+	// only read the sync unit's checkpoint information
+	syncCheckpointTable := fmt.Sprintf("%s_syncer_checkpoint", subTaskCfg.Name)
+	exist, err := utils.TableExists(db.DB, subTaskCfg.MetaSchema, syncCheckpointTable)
+	if err != nil {
+		return nil, err
+	}
+	if !exist {
+		return nil, nil
+	}
+
+	query := fmt.Sprintf("SELECT `binlog_name`, `binlog_pos` FROM %s.%s where id = ?", subTaskCfg.MetaSchema, syncCheckpointTable)
```

Comment:
Is it enough to just get the _global_ checkpoint?

Reply:
yes, I use checkpoint.GlobalPoint in 69363d7